rules/prefer-top-level-await.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+const {findVariable, getFunctionHeadLocation} = require('eslint-utils');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+const {matches, memberExpressionSelector} = require('./selectors');
+
+const ERROR_PROMISE = 'promise';
+const ERROR_IIFE = 'iife';
+const ERROR_IDENTIFIER = 'identifier';
+const SUGGESTION_ADD_AWAIT = 'add-await';
+const messages = {
+	[ERROR_PROMISE]: 'Prefer top-level await over using promise.',
```

Comment:
```suggestion
	[ERROR_PROMISE]: 'Prefer top-level await over using a promise.',
```

But maybe we should instead say:

```suggestion
	[ERROR_PROMISE]: 'Prefer top-level await over using a promise chain.',
```